pkg/postgres/pgutils/retry.go

```diff
@@ -0,0 +1,37 @@
+package pgutils
+
+import (
+	"time"
+)
+
+const (
+	interval = 5 * time.Second
+	timeout  = 5 * time.Minute
+)
+
+// RetryQuery is used to specify how long to retry to successfully run a query
+// that fails with Transient errors
+func RetryQuery(fn func() error) error {
```

Comment:
Looking at the use of `RetryQuery`, I think you could benefit from generics:
```
func RetryQuery[T any](fn func() (T, error)) {
```
to avoid the extra `var` declaration outside of the closure

Reply:
oh yeah, this is so nice :)